src/actions/rootRequestedThunk.tsx

```diff
@@ -23,15 +23,12 @@ export const rootRequestedThunk = () => {
         const handleTheData = (dbrecords: any) => {
             const convert = (dbrecord: IDatabaseRecord) => {
                 return {
-                    parent:         -1,
+                    parent:         dbrecord.childof,
```

Comment:
not sure what you mean with "in the preprocessing" here, 

(and I also dont see whhy we need a middle ground while my opinion is clearly better here) /sarcasm